File: tests/fixtures/sample.cs
using System;
using System.Collections.Generic;

namespace DraculaColorful;

public sealed record ThemeToken<TValue>(string Name, TValue Value);

public class ThemeRenderer
{
    private static readonly string DefaultAccent = "#BD93F9";
    private readonly List<ThemeToken<string>> _tokens = new();

    public string Accent { get; init; } = DefaultAccent;

    [Obsolete]
    public string RenderToken(string name, Func<string, string> formatter)
    {
        var token = new ThemeToken<string>(name, Accent);
        _tokens.Add(token);
        return formatter(token.Value);
    }
}

File: tests/fixtures/sample.rs
pub struct ThemeToken<TValue> {
    pub name: String,
    pub value: TValue,
}

impl<TValue> ThemeToken<TValue> {
    pub const DEFAULT_ACCENT: &'static str = "#BD93F9";

    pub fn new(name: String, value: TValue) -> Self {
        Self { name, value }
    }
}

pub fn render_token(name: &str) -> String {
    let token = ThemeToken::new(name.to_owned(), ThemeToken::<String>::DEFAULT_ACCENT.to_owned());
    format!("{}", token.value)
}
